engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/ExternalTaskManager.java

```diff
@@ -73,25 +74,27 @@ public List<ExternalTaskEntity> findExternalTasksByProcessInstanceId(String proc
   }
 
   @SuppressWarnings("unchecked")
-  public List<ExternalTaskEntity> selectExternalTasksForTopics(Collection<TopicFetchInstruction> queryFilters, int maxResults, boolean usePriority) {
+  public List<ExternalTaskEntity> selectExternalTasksForTopics(Collection<TopicFetchInstruction> queryFilters,
+                                                               int maxResults,
+                                                               boolean usePriority,
+                                                               boolean useCreateTime,
+                                                               Direction createTimeDirection) {
```

Comment:
🔧 Should we instead introduce a collection for sort options? This would allow us to introduce more sorting options in the future without just adding two properties for each option to the method.

Reply:
The sorting list parameter for the API is a reasonable design decision cause it's public API and introduced once.
In this case, the `ExternalTaskManager` is implementation detail and _can_ change. In this regard, i'd prefer to go YAGNI and make the refactoring when needed (e.g when the second sorting property is introduced)

Reply:
That is a fair point; I leave this up to you. If we use the more generic approach everywhere else, I would also use it here. This is a less complex solution, in my opinion.